rest/route/service.go

```diff
@@ -9,103 +9,119 @@ import (
 
 const defaultLimit = 100
 
+type HandlerOpts struct {
+	App               *gimlet.APIApp
+	APIQueue          amboy.Queue
+	SingleWorkerQueue amboy.Queue
+	URL               string
+	SuperUsers        []string
+	GithubSecret      []byte
+}
+
 // AttachHandler attaches the api's request handlers to the given mux router.
 // It builds a Connector then attaches each of the main functions for
 // the api to the router.
-func AttachHandler(app *gimlet.APIApp, queue amboy.Queue, URL string, superUsers []string, githubSecret []byte) {
+func AttachHandler(opts HandlerOpts) {
 	sc := &data.DBConnector{}
 
-	sc.SetURL(URL)
-	sc.SetSuperUsers(superUsers)
+	sc.SetURL(opts.URL)
+	sc.SetSuperUsers(opts.SuperUsers)
 
 	// Middleware
 	superUser := gimlet.NewRestrictAccessToUsers(sc.GetSuperUsers())
 	checkUser := gimlet.NewRequireAuthHandler()
 	addProject := NewProjectContextMiddleware(sc)
 	checkProjectAdmin := NewProjectAdminMiddleware(sc)
 
-	settings := evergreen.GetEnvironment().Settings()
+	env := evergreen.GetEnvironment()
+	settings := env.Settings()
+
+	// Support override for testing
+	if opts.SingleWorkerQueue == nil {
+		opts.SingleWorkerQueue = env.SingleWorkerQueue()
+	}
```

Comment:
just fix the tests?

put settings or the things you need from settings in the Handler opts

Reply:
Done, I think? I'm not sure about if putting the queue in `GetRouter` only for apiRestV2 is the right thing, but seems like.